instana/resource-synthetic-location.go

```diff
@@ -0,0 +1,96 @@
+package instana
+
+import (
+	"fmt"
+
+	"github.com/gessnerfl/terraform-provider-instana/instana/restapi"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// NewSyntheticLocationDataSource creates a new DataSource for Synthetic Locations
+func NewSyntheticLocationDataSource() DataSource {
+	return &syntheticLocationDataSource{}
+}
+
+// ResourceInstanaSyntheticLocation the name of the terraform-provider-instana resource to manage synthetic locations
+const ResourceInstanaSyntheticLocation = "instana_synthetic_location"
+
+const (
+	//SyntheticLocationFieldLabel constant value for the schema field label
+	SyntheticLocationFieldLabel = "label"
+	//SyntheticLocationFieldDescription constant value for the computed schema field description
+	SyntheticLocationFieldDescription = "description"
+	//SyntheticLocationFieldLocationType constant value for the schema field location_type
+	SyntheticLocationFieldLocationType = "location_type"
+	//DataSourceSyntheticLocation the name of the terraform-provider-instana data sourcefor synthetic location specifications
+	DataSourceSyntheticLocation = "instana_synthetic_location"
+)
+
+type syntheticLocationDataSource struct{}
+
+// CreateResource creates the resource handle Synthetic Locations
+func (ds *syntheticLocationDataSource) CreateResource() *schema.Resource {
+	return &schema.Resource{
+		Read: ds.read,
+		Schema: map[string]*schema.Schema{
+			SyntheticLocationFieldLabel: {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Friendly name of the Synthetic Location",
+				ValidateFunc: validation.StringLenBetween(0, 512),
+			},
+			SyntheticLocationFieldDescription: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The description of the Synthetic location",
+				ValidateFunc: validation.StringLenBetween(0, 512),
+			},
+			SyntheticLocationFieldLocationType: {
```

Comment:
Would it make sense to restrict the type to Public and Private or are there more values supported